tests/components/pterodactyl/test_config_flow.py

```diff
@@ -0,0 +1,160 @@
+"""Test the Pterodactyl config flow."""
+
+from pydactyl import PterodactylClient
+from pydactyl.exceptions import ClientConfigError, PterodactylApiError
+from pydactyl.responses import PaginatedResponse
+import pytest
+
+from homeassistant.components.pterodactyl.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import (
+    TEST_SERVER,
+    TEST_SERVER_LIST_DATA,
+    TEST_SERVER_UTILIZATION,
+    TEST_URL,
+    TEST_USER_INPUT,
+)
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("mock_pterodactyl")
+async def test_full_flow(
+    hass: HomeAssistant, mock_pterodactyl: PterodactylClient
+) -> None:
+    """Test full flow without errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
```

Comment:
we should also patch `async_setup_entry` search for `mock_setup_entry` for easy copy paste